recipes/fruit/all/conanfile.py

```diff
@@ -54,69 +58,55 @@ def configure(self):
-
-    @property
-    def _extracted_dir(self):
-        return self.name + "-" + self.version
-
-    def _get_source(self):
+    def source(self):
         if Version(self.version) == "3.4.0":
             filename = os.path.basename(self.conan_data["sources"][self.version]["url"])
-            tools.download(filename=filename, **self.conan_data["sources"][self.version])
+            download(self, filename=filename, **self.conan_data["sources"][self.version])
+            extracted_dir = self.name + "-" + self.version
 
             with tarfile.TarFile.open(filename, 'r:*') as tarredgzippedFile:
                 # NOTE: In fruit v3.4.0, The archive file contains the file names
                 # build and BUILD in the extras/bazel_root/third_party/fruit directory.
                 # Extraction fails on a case-insensitive file system due to file
                 # name conflicts.
                 # Exclude build as a workaround.
-                exclude_pattern = "%s/extras/bazel_root/third_party/fruit/build" % (self._extracted_dir,)
+                exclude_pattern = f"{extracted_dir}/extras/bazel_root/third_party/fruit/build"
                 members = list(filter(lambda m: not fnmatch(m.name, exclude_pattern),
                                     tarredgzippedFile.getmembers()))
-                tarredgzippedFile.extractall(".", members=members)
+                tarredgzippedFile.extractall(path=self.source_folder, members=members)
+            allfiles = os.listdir(os.path.join(self.source_folder, extracted_dir))
+            for file_name in allfiles:
+                shutil.move(os.path.join(self.source_folder, extracted_dir, file_name), self.source_folder)
         else:
-            tools.get(**self.conan_data["sources"][self.version])
-
-    def source(self):
-        self._get_source()
-
-        os.rename(self._extracted_dir, self._source_subfolder)
-
-    def _configure_cmake(self):
-        if not self._cmake:
-            self._cmake = CMake(self)
-            self._cmake.definitions["FRUIT_USES_BOOST"] = self.options.use_boost
-            self._cmake.definitions["FRUIT_ENABLE_COVERAGE"] = False
-            self._cmake.definitions["RUN_TESTS_UNDER_VALGRIND"] = False
-            self._cmake.definitions["FRUIT_ENABLE_CLANG_TIDY"] = False
-
-            self._cmake.configure(build_folder=self._build_subfolder)
-        return self._cmake
-
-    def _patch_files(self):
-        if self.version in self.conan_data["patches"]:
-            for patch in self.conan_data["patches"][self.version]:
-                tools.patch(**patch)
+            get(self, **self.conan_data["sources"][self.version], strip_root=True)
+
+    def generate(self):
+        tc = CMakeToolchain(self)
+        tc.cache_variables["FRUIT_USES_BOOST"] = self.options.use_boost
+        tc.variables["FRUIT_ENABLE_COVERAGE"] = False
+        tc.variables["RUN_TESTS_UNDER_VALGRIND"] = False
+        tc.variables["CMAKE_CXX_STANDARD"] = 11
```

Comment:
@uilianries Do we need `BUILD_SHARED_LIBS=False`, if `self.options.shared==False` here? Or does CMakeToolchain it automatically?

Reply:
@tiolan It's automatically managed: https://docs.conan.io/en/latest/reference/conanfile/tools/cmake/cmaketoolchain.html#extending-and-customizing-cmaketoolchain